### ISSUE TITLE
Update debian package generation to comply with current Policy

### ISSUE THREAD

**Issue by dankamongmen** [ros-infrastructure/bloom#532]
See PR #531. There are a few ways in which bloom-generated debs depart from Debian Policy, resulting primarily in Lintian warnings, but also possibly unexpected behavior. It would be good to bring things into conformance with current Policy, though as @nuclearsandwich points out, it's not worth breaking older installs.

**Comment by mikepurvis** [comment]
Is there a way to silence the installed-into-opt warning? With that one spamming on every single file that a bloom-generated package installs, it's hard to even see what the other lintian complaints are.

**Comment by tfoote** [comment]
It looks like you can override lintian errors as documented here: https://lintian.debian.org/manual/section-2.4.html  

Other lintian errors sampling a random build for source debs look like http://build.ros.org/view/Msrc_uB/job/Msrc_uB__actionlib_msgs__ubuntu_bionic__source/3/console
```
15:26:41 W: ros-melodic-actionlib-msgs source: no-debian-copyright
15:26:41 W: ros-melodic-actionlib-msgs source: ancient-standards-version 3.9.2 (released 2011-04-07) (current is 4.1.4)
```

The copyright file I think was resolved recently: https://github.com/ros-infrastructure/bloom/pull/470

The `Standards-Version` is embedded in our control file templates and likely could be updated. But would need some QA to understand what we're changing but could likely be pulled forward. It has probably just been untouched since we first started generating debian packages.

https://github.com/ros-infrastructure/bloom/search?utf8=%E2%9C%93&q=Standards-Version&type=

**Comment by dankamongmen** [comment]
You don't want to bump the Standards-Version until you've verified that you've conformed to (or at least consciously rejected) the new elements of Debian Policy. That's what the field serves to indicate: "As the package maintainer, I believe this package to conform to Debian Policy through at least the specified version." I'm planning on working through up to current 4.3.0.4 this weekend if possible. 

I think we're just gonna have to elide the /opt warnings, assuming ROS is wedded to /opt.

**Comment by tfoote** [comment]
> You don't want to bump the Standards-Version until you've verified that you've conformed to (or at least consciously rejected) the new elements of Debian Policy. That's what the field serves to indicate: "As the package maintainer, I believe this package to conform to Debian Policy through at least the specified version." I'm planning on working through up to current 4.3.0.4 this weekend if possible.

Certainly we'll need to check that we're compliant. That would be great if you have a chance to review it!

> I think we're just gonna have to elide the /opt warnings, assuming ROS is wedded to /opt.

Yeah, debian has gotten pretty strict on this. This is an explicit choice to diverge from the debian guidelines. This allows us to support multiple distros installed side by side. It's technically following the FHS as we're a registered [LANANA provider](http://www.lanana.org/lsbreg/providers/providers.txt) for `ros`.

**Comment by dankamongmen** [comment]
@tfoote  absolutely, just as @mikepurvis says above, i think it would be good to elide the /opt warnings, as they do tend to represent most of the lintian mass. The Override method you linked to ought suffice -- I wasn't aware of that, nice find!

**Comment by realtime-neil** [comment]
Does the merge of #531 justify the closing of this issue, and if not, then what remains?

**Comment by dankamongmen** [comment]
@realtime-neil it would be good at a minimum to put the Override mentioned by @tfoote into play, so that other Lintian warnings are more readily apparent.

**Comment by dankamongmen** [comment]
Well, just wasted two hours only to find out that the `dir-or-file-in-opt` lintian error cannot be overrode:

```
E: reach-rs-driver: dir-or-file-in-opt opt/ros/
N: 
N:    Debian packages should not install into /opt, because it is reserved for
N:    add-on software.
N:    
N:    Refer to Filesystem Hierarchy Standard (/opt : Add-on application
N:    software packages) for details.
N:    
N:    Severity: serious, Certainty: certain
N:    
N:    Check: files, Type: binary, udeb
N: 
E: reach-rs-driver: dir-or-file-in-opt opt/ros/melodic/
E: reach-rs-driver: dir-or-file-in-opt opt/ros/melodic/lib/
E: reach-rs-driver: dir-or-file-in-opt ... use --no-tag-display-limit to see all (or pipe to a file/program)
...
N: Some overrides were ignored, since the tags were marked "non-overridable".
N: Use --verbose for more information.
```

[0001.lintian.override.txt](https://github.com/ros-infrastructure/bloom/files/3540349/0001.lintian.override.txt)

I've attached the diff that led to this output. Lame.

Why *does* ROS feel the need to install to /opt?

**Comment by realtime-neil** [comment]
@dankamongmen I've heard it said that `/opt/ros/${ROS_DISTRO}` is because supporting multiple `${ROS_DISTRO}` is good. This kind of we-support-everything attitude plays all kinds of havoc with that new-fangled curio, the linux distribution package manager.
